File: daily_question/1544._make_the_string_great/javascript/solution.js
/**
 * @param {string} s
 * @return {string}
 */
var makeGood = function (s) {
    stack = [];
    for (let char of s) {
        if (stack.length > 0 && Math.abs(char.charCodeAt(0) - stack[stack.length - 1].charCodeAt(0)) === 32) {
            stack.pop();
            continue;
        }
        stack.push(char);
    }
    return stack.join('');
};

console.log(makeGood("leEeetcode") === "leetcode");
console.log(makeGood("abBAcC") === "");
console.log(makeGood("s") === "s");

File: daily_question/1544._make_the_string_great/ruby/solution.rb
# @param {String} s
# @return {String}
def make_good(s)
    stack = []
    s.each_char do |char|
        if !stack.empty? && (char.ord - stack[-1].ord).abs == 32
            stack.pop
            next
        end
        stack.push(char)
    end
    stack.join('')
end

puts make_good("leEeetcode") == "leetcode"
puts make_good("abBAcC") == ""
puts make_good("s") == "s"